### ISSUE TITLE
Consider releasing a new greenlet 1.1.3 version compatible with Python 3.11

### ISSUE THREAD

**Issue by vstinner** [python-greenlet/greenlet#305]
The current main branch is a new greenlet 2.0 version which has a different API than greenlet 1.x and is written in C++. Shipping greenlet 2.0 alpha in the future Fedora 37 scheduled for next October (2022) sounds risky. We (Fedora) would like to upgrade Python to Python 3.11 but keep greenlet 1.x.

Would you be open to consider releasing a new greenlet 1.1.3 version which would be compatible with Python 3.11?

We prepared a PR on top on the 1.1.2 version to add Python 3.11 support.

Python 3.11 final version is also scheduled next October (2022): https://peps.python.org/pep-0664/#schedule

See also the issue #303: "Schedule for a beta release of greenlet 2.0?"

**Comment by vstinner** [comment]
We wrote the PR #306 which adds Python 3.11 support to greenlet 1.x.

**Comment by vstinner** [comment]
Downstream patch in Fedora: https://src.fedoraproject.org/rpms/python-greenlet/pull-request/3

**Comment by vstinner** [comment]
@jamadden: What do you think of this issue?

It seems like @ssbarnea also wants a greenlet compatible with Python 3.11: https://github.com/python-greenlet/greenlet/pull/306#issuecomment-1152956296

**Comment by hugovk** [comment]
Would be great to get a release out, as it there's been some stability issues with Python 3.11 beta so it's very important to test it, and this is blocking other projects testing 3.11 that depend on greenlet. 

See:

* https://discuss.python.org/t/the-last-3-11-beta-release-3-11-0b5-is-now-available/17693

Thank you!

**Comment by changyuheng** [comment]
Python 3.11 rc1 will be released on [August 1st](https://peps.python.org/pep-0664/). It would be great if we can have the Python 3.11 support at the same time.

**Comment by jamesob** [comment]
Would really appreciate a fix here; `pynvim` is dependent on greenlet, and so many nvim configurations are now broken when using Python 3.11.

**Comment by changyuheng** [comment]
> Would really appreciate a fix here; `pynvim` is dependent on greenlet, and so many nvim configurations are now broken when using Python 3.11.

Exactly. I just switched to [nvim-cmp](https://github.com/hrsh7th/nvim-cmp) from [coq_nvim](https://github.com/ms-jpq/coq_nvim) for this reason yesterday.

**Comment by jamadden** [comment]
#306 has been merged and 1.1.3 released to PyPI.

**Comment by FrankJian** [comment]
Hello , 
  today i met the issue when install the 1.1.3 in the docker (base image is `python:3.8.12-alpine3.14`)
error log below when do pip install 
```
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [111 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-38
      creating build/lib.linux-x86_64-cpython-38/greenlet
      copying src/greenlet/__init__.py -> build/lib.linux-x86_64-cpython-38/greenlet
      creating build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/__init__.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_contextvars.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_cpp.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_extension_interface.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_gc.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_generator.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_generator_nested.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_greenlet.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_leaks.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_stack_saved.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_throw.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_tracing.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_version.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/test_weakref.py -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      running egg_info
      writing src/greenlet.egg-info/PKG-INFO
      writing dependency_links to src/greenlet.egg-info/dependency_links.txt
      writing requirements to src/greenlet.egg-info/requires.txt
      writing top-level names to src/greenlet.egg-info/top_level.txt
      reading manifest file 'src/greenlet.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      no previously-included directories found matching 'docs/_build'
      warning: no files found matching '*.py' under directory 'appveyor'
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '*.pyd' found anywhere in distribution
      warning: no previously-included files matching '*.so' found anywhere in distribution
      warning: no previously-included files matching '.coverage' found anywhere in distribution
      adding license file 'LICENSE'
      adding license file 'LICENSE.PSF'
      adding license file 'AUTHORS'
      writing manifest file 'src/greenlet.egg-info/SOURCES.txt'
      /usr/local/lib/python3.8/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'greenlet.platform' as data is deprecated, please list it in `packages`.
          !!
      
      
          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'greenlet.platform' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
      
          'greenlet.platform' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).
      
          Please make sure that 'greenlet.platform' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
      
          You can read more about "package discovery" and "data files" on setuptools
          documentation page.
      
      
      !!
      
        check.warn(importable)
      copying src/greenlet/greenlet.c -> build/lib.linux-x86_64-cpython-38/greenlet
      copying src/greenlet/greenlet.h -> build/lib.linux-x86_64-cpython-38/greenlet
      copying src/greenlet/slp_platformselect.h -> build/lib.linux-x86_64-cpython-38/greenlet
      creating build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/setup_switch_x64_masm.cmd -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_aarch64_gcc.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_alpha_unix.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_amd64_unix.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_arm32_gcc.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_arm32_ios.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_csky_gcc.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_m68k_gcc.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_mips_unix.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_ppc64_aix.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_ppc64_linux.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_ppc_aix.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_ppc_linux.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_ppc_macosx.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_ppc_unix.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_riscv_unix.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_s390_unix.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_sparc_sun_gcc.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_x32_unix.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_x64_masm.asm -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_x64_masm.obj -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_x64_msvc.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_x86_msvc.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/platform/switch_x86_unix.h -> build/lib.linux-x86_64-cpython-38/greenlet/platform
      copying src/greenlet/tests/_test_extension.c -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      copying src/greenlet/tests/_test_extension_cpp.cpp -> build/lib.linux-x86_64-cpython-38/greenlet/tests
      running build_ext
      building 'greenlet._greenlet' extension
      creating build/temp.linux-x86_64-cpython-38
      creating build/temp.linux-x86_64-cpython-38/src
      creating build/temp.linux-x86_64-cpython-38/src/greenlet
      gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -DTHREAD_STACK_SIZE=0x100000 -fPIC -I/usr/local/include/python3.8 -c src/greenlet/greenlet.c -o build/temp.linux-x86_64-cpython-38/src/greenlet/greenlet.o
      gcc -shared -Wl,--strip-all build/temp.linux-x86_64-cpython-38/src/greenlet/greenlet.o -L/usr/local/lib -o build/lib.linux-x86_64-cpython-38/greenlet/_greenlet.cpython-38-x86_64-linux-gnu.so
      building 'greenlet.tests._test_extension' extension
      creating build/temp.linux-x86_64-cpython-38/src/greenlet/tests
      gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -DTHREAD_STACK_SIZE=0x100000 -fPIC -Isrc/greenlet/ -I/usr/local/include/python3.8 -c src/greenlet/tests/_test_extension.c -o build/temp.linux-x86_64-cpython-38/src/greenlet/tests/_test_extension.o
      gcc -shared -Wl,--strip-all build/temp.linux-x86_64-cpython-38/src/greenlet/tests/_test_extension.o -L/usr/local/lib -o build/lib.linux-x86_64-cpython-38/greenlet/tests/_test_extension.cpython-38-x86_64-linux-gnu.so
      building 'greenlet.tests._test_extension_cpp' extension
      gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -DTHREAD_STACK_SIZE=0x100000 -fPIC -Isrc/greenlet/ -I/usr/local/include/python3.8 -c src/greenlet/tests/_test_extension_cpp.cpp -o build/temp.linux-x86_64-cpython-38/src/greenlet/tests/_test_extension_cpp.o
      gcc: fatal error: cannot execute 'cc1plus': execvp: No such file or directory
      compilation terminated.
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for greenlet
```

and i downgrade to version 1.1.2, everything are fine

**Comment by vstinner** [comment]
> #306 has been merged and 1.1.3 released to PyPI.

Thanks a lot @jamadden!
